### PR TITLE
Default code page fix +  crash fix

### PR DESCRIPTION
**Pull request by MikhailArkhipov**
Leave code page alone, R sets it correctly on its own (tried on Spanish Windows Language pack)
Fix crash in parameter-less signatures

### CODE REVIEW

File: src/Host/Client/Impl/Session/RSessionEvaluationCommands.cs [hunk 1]
```diff
@@ -222,11 +222,13 @@ public static Task OptionsSetWidthAsync(this IRExpressionEvaluator evaluation, i
 
         public static async Task SetCodePageAsync(this IRExpressionEvaluator evaluation, int codePage, CancellationToken cancellationToken = default(CancellationToken)) {
             var cp = $".{codePage}";
-            if (codePage == 0) {
-                cp = await evaluation.IsRSessionPlatformWindowsAsync() ? ".437" : "en_US.UTF-8";
+            if (codePage == 0 && !await evaluation.IsRSessionPlatformWindowsAsync(cancellationToken)) {
+                cp = "en_US.UTF-8";
```
Comment (int19h) [inline src/Host/Client/Impl/Session/RSessionEvaluationCommands.cs]: I'm confused about how this works. So now if this function gets called with codePage==0, we try to set LC_CTYPE to `".0"` - but that's not a valid codepage, is it?

Comment (MikhailArkhipov) [inline src/Host/Client/Impl/Session/RSessionEvaluationCommands.cs]: Whoops, wrong check...

**Comment by int19h** [comment]
I wonder why we we changing the default in the first place - surely that was for some encoding-related reason. Does it do the right thing now with e.g. Russian?

**Comment by MikhailArkhipov** [comment]
I set Spanish language pack and R locale was as user wanted. Tried setwd-getwd with Unicode path (Russian) on Spanish UI? worked fine. I think we used to set OEM b/c technically R is non-Unicode program. OEM should work most of the time, I think. I can try Russian lang pack too.

**Comment by MikhailArkhipov** [comment]
Russian works fine, publishing markdown to folder named in Unicode works fine too (we switched to short paths anyways a while ago). I think the only think OEM might affect is reading/writing files without specifying encoding, but it is up to the user to specify correct one.

**Pull request merged**